src/legacy/core_plugins/table_vis/public/agg_table/agg_table.js

```diff
@@ -94,16 +96,16 @@ uiModules
           }).join('');
         };
 
-        $scope.$watch('table', function () {
-          const table = $scope.table;
+        $scope.$watchMulti(['table', 'exportTitle', 'showPercentage', 'percentageCol', 'totalFunc'], function () {
```

Comment:
Why we are applying a multi watch to all these variables?
If before this patch, we were just only watching `table` shouldn't we watch table + only the variable that we added like `showPercentage` and `percentageCol`?

A multiwatch like this means that this function is called whenever any of these variable is changed, is really required?

Reply:
`exportTitle` and `totalFunc` were already in use in the callback function.
`showPercentage` and `percentageCol` I added for the work I'm adding here.

All four of these properties I added because they're used to compute something within the callback. Right now, it just so happens that the table variable changes at the same time as those change and they don't change independently. But it seems like a place a bug could sneak in if somewhere else there was a refactor that let you update the `exportTitle` (or anything else) independently of the table. This way, if the way any of these variables that get passed in change, everything will continue to "Just Work™️" (ideally).

I thought this was more explicit about the real dependencies of the function.